### DISCUSSION TITLE
[HELP] Connecting to GROWATT SPH1000 TL BH-UP Inverter

### DISCUSSION THREAD

**Discussion by Rasmusbangp** [wills106/homeassistant-solax-modbus#497] [Q&A]
First of all, im sorry if this gets a bit messy and i will of cause try to provide all info needed, if helpfull team viewer is also possible if required from your site.

Really Hope some can help me with this integration and make it run on my growatt sph10ktl. 

Im trying to connect, but apparently im getting no info at all, only a short blue light when i try to connect to the waveshare address via HA.

Waveshare RS485 to Eth PoE (Firmware 1.486):
Baud Rate: 9600
Work Mode: TCP Server
Transfer Protocol: Modbus_TCP Protocol
Modbus Gateway Type: Simple Modbus TCP to RTU

![Skærmbillede 2023-08-05 kl  20 49 51](https://github.com/wills106/homeassistant-solax-modbus/assets/134646879/4cf653f8-1d34-4c28-b3d0-ca2020885bc4)

**Home Assistant:**
Interface: TCP / Ethernet
Modbus Address: 1
Inverter Type: Growatt
(EPS, Dry Contact Box and Parallel Mode unchecked)
IP Address: 192.168.68.111
TCP Port: 502
Modbus Variant: Modbus RTU over TCP (Ive also tried Modbus TCP)

**Mandatory details**
1. Integration Version: 23.08.3
2. HA Version: 23.7.2
3. Inverter Brand: Growatt SPH1000 TL BH-UP HYBRID INVERTER WITH 10KWH BATTERY BRAND EITAI
4. SERIAL: V008
5. MODEL: GT1103F8AD
6. FW: YA1.0
7. Plugin Used: [homeassistant-solax-modbus]

**Detailed Error Log**
Denne fejl stammer fra en brugerdefineret integration.
Logger: custom_components.solax_modbus.plugin_growatt
Source: custom_components/solax_modbus/plugin_growatt.py:3110 
Integration: SolaX Inverter Modbus (documentation, issues) 
First occurred: 20.44.32 (1 occurrences) 
Last logged: 20.44.32

unrecognized GROWATT SPH10KTL inverter type - firmware version : unknown

Denne fejl stammer fra en brugerdefineret integration.
Logger: custom_components.solax_modbus.plugin_growatt
Source: custom_components/solax_modbus/plugin_growatt.py:3091 
Integration: SolaX Inverter Modbus (documentation, issues) 
First occurred: 20.44.32 (1 occurrences) 
Last logged: 20.44.32

GROWATT SPH10KTL: cannot find firmware version, even not for other Inverter

Denne fejl stammer fra en brugerdefineret integration.
Logger: custom_components.solax_modbus.plugin_growatt
Source: custom_components/solax_modbus/plugin_growatt.py:3089 
Integration: SolaX Inverter Modbus (documentation, issues) 
First occurred: 20.44.32 (1 occurrences) 
Last logged: 20.44.32

GROWATT SPH10KTL: reading firmware number from address 0xbb9 failed; other address may succeed

Denne fejl stammer fra en brugerdefineret integration.
Logger: custom_components.solax_modbus.plugin_growatt
Source: custom_components/solax_modbus/plugin_growatt.py:3086 
Integration: SolaX Inverter Modbus (documentation, issues) 
First occurred: 20.44.27 (1 occurrences) 
Last logged: 20.44.27

GROWATT SPH10KTL: reading firmware number from address 0x9 failed; other address may succeed

If there are no log's showing for SolaX try the following depending on your language settings, you can find the full logs under:

[home-assistant_2023-08-05T18-45-27.441Z.log](https://github.com/wills106/homeassistant-solax-modbus/files/12267965/home-assistant_2023-08-05T18-45-27.441Z.log)

**Additional context**
Cable is connected blue to A and Blue/White to B at the waveshare and the other end is in port 485-1
Waveshare running on 16V power supply, with red light in PWR, Blue in ACT and green/yellowish in LINK (Link turns blue for few seconds, when i try to connect to the ip-adress 192.168.68.111 via Sole Inverter Modbus from HA.

Device settings from virCom.
![Skærmbillede 2023-08-05 kl  22 25 00](https://github.com/wills106/homeassistant-solax-modbus/assets/134646879/0262e761-64b9-4f7d-9faf-2a1a2d68febc)

Advanced settings from VirCom
![Skærmbillede 2023-08-05 kl  22 25 11](https://github.com/wills106/homeassistant-solax-modbus/assets/134646879/ff4c201d-fc02-4051-8137-c81c045fb1aa)

I hope some can guide me in the right direction :)

Thank u all :)

**Comment by wills106** [comment]
Moved to Discussions as not an Integration issue.
